share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareGroupOffset.java

```diff
@@ -245,4 +246,15 @@ public String toString() {
             ", stateBatches=" + stateBatches +
             '}';
     }
+
+    public Builder builderSupplier() {
+        return new Builder()
+            .setSnapshotEpoch(snapshotEpoch)
+            .setStateEpoch(stateEpoch)
+            .setLeaderEpoch(leaderEpoch)
+            .setStartOffset(startOffset)
+            .setStateBatches(new ArrayList<>(stateBatches))
```

Comment:
@smjn, one question, why `stateBatches` needs to be wrapped with `ArrayList` here? Thank you.